pandas/tests/indexing/test_scalar.py

```diff
@@ -136,30 +136,48 @@ def test_at_to_fail(self):
         result = s.at["a"]
         assert result == 1
         msg = (
-            "At based indexing on an non-integer index can only have "
-            "non-integer indexers"
+            "cannot do label indexing on <class 'pandas.core.indexes.base.Index'> "
```

Comment:
Can you parametrize this test for at / loc? I think would make it easier and more comprehensive

Reply:
looking at this, im definitely going to split this into more targeted tests, will look at parametrizing; might benefit from a helper to assert we get the same behavior

Reply:
refactored to make targeted tests, this file needs a dedicated pass to move old testing patterns to parametrization